File: labs_26_04_18/ex1/Rectangle.java
package labs_26_04_18.ex1;

// Klasa dla Prostokąta
class Rectangle extends Quadrangle {
    private final double width;
    private final double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    @Override
    public double calculateArea() {
        return width * height;
    }

    @Override
    public double calculatePerimeter() {
        return 2 * width + 2 * height;
    }

    @Override
    public String getName() {
        return "Prostokąt";
    }
}

File: labs_26_04_18/ex2/src/figures/rectangle.cpp
#include "../../inc/rectangle.hpp"

Rectangle::Rectangle(double w, double h) : width(w), height(h) {}
double Rectangle::calculateArea() const { return width * height; }
double Rectangle::calculatePerimeter() const { return 2 * width + 2 * height; }
std::string Rectangle::getName() const { return "Prostokat"; }